index.rst

```diff
@@ -42,20 +42,163 @@
+The commands will indicate which data to process and the name of the pipeline to execute but they will not include algorithmic code.
+The data will be specified using a "dataId" to match standard Data Butler usage.
+Once the pipeline completes an event will be published.
+
+.. [*] In :cite:`LDM-148` this is called the "OCS-Controlled Batch Service."
+
+The sequence of events for a typical observing script would then be:
+
+* Send a command to the camera to take an image.
+  This command should include the relevant groupId string as supplied from the ScriptQueue, and the observing type (such as science, bias, or flat).
+* Wait for the event indicating that the exposure has been written to the OODS (see :cite:`DMTN-111`).
+  This event will include the relevant obsid.
+* Send a command to the OCPS requesting that this obsid be processed.
+  This command should include the name of the pipeline to execute.
+* Acquire more data and send processing requests for each one.
+* When all data are acquired and single-frame processing has completed, send a command to the OCPS requesting that a co-add pipeline be executed with all data using the shared groupId.
+* An observing script can finish as soon as it has sent the final co-add command to the OCPS.
+
+In some cases a complex script might require multiple groups to be defined during execution.
+To deal with this case Python observing scripts shall be given an API to generate a new unique group ID from the value supplied to the script from the ScriptQueue.
+
+.. warning::
+
+   Observing scripts for standard visits and alternate standard visits should never use a group ID that differs from that supplied by the ScriptQueue.
+   This is because the prompt processing system uses the group ID to collate data and expects a specific value to appear that matches the value published by the ScriptQueue.
+
+As pipelines execute, they will be able to calculate data quality assessment metrics and other results of interest such as centroids.
+These can be published as telemetry to allow them to be displayed to the observer.
+This telemetry can be received by the observing script that issued the data processing request, allowing the script to fine tune its observing strategy.
+These metrics should be associated with the history of the exposure in the same way as observer comments are associated with specific observations.
+
+.. note::
+
+   Notebooks can also block waiting for the processed data events to arrive before doing bespoke commissioning visualization or further detailed analysis that is decoupled from observing.
+
+The OCPS CSC
+------------
+
+The OCPS CSC will support one command, let's say ``processData``, used by observing scripts.
+The command will need to take at least a dataId and a pipeline name, and can possibly also take configuration overrides.
+The dataId looks like a ``dict`` and would be expected to include either an obsid or a group ID and possibly a detector number.
```

Comment:
This is similar to [DM-22496](https://jira.lsstcorp.org/browse/DM-22496); if a serialization is defined there, reusing it in this context should at least be considered.  If a single OCPS deployment is to be used with multiple instruments (LATISS, ComCam, LSSTCam), it may be necessary to specify the repository, although the dataset type can likely be fixed.